### ISSUE TITLE
[CP][beta channel][stable channel] Fix

### ISSUE THREAD

**Issue by mraleph** [dart-lang/sdk#46381]
**commit(s) to merge**: 4c1a153c090e6c99a30fdfa0f2e0dbb34f15df7d

**merge instructions**: clean merge

**What is the issue**: Under some circumstances encountering compile time error in user code would cause front-end to crash

**What is the fix**: When attempting to report compile time error ensure that location information is present on the problem node. 

**Why cherrypick**: Leads to crashes encountered by Flutter users, e.g. flutter/flutter#84212. Resulting crashes are impossible for users to diagnose and work around.

**Risk**: very low, one line change.

**Link to original  issue(s)**: flutter/flutter#84212

/cc @dgrove @kevmoo @mit-mit @whesse @athomas @vsmenon @franklinyow

**Comment by johnniwinther** [comment]
LGTM

**Comment by sigmundch** [comment]
lgtm

**Comment by athomas** [comment]
Given the LGTM, I'm going to assume this has been approved for beta.

**Comment by athomas** [comment]
Merged to beta in bd5cd8a65baf2cdad8f5ba9b350c4e912d2a6cdf (`2.14.0-188.3.beta`). Issue will remain open until merged to stable.

**Comment by athomas** [comment]
Not a clean merge:
```
$ git diff
diff --cc pkg/front_end/lib/src/fasta/source/source_library_builder.dart
index 097bde39876,8736d5a9741..00000000000
--- a/pkg/front_end/lib/src/fasta/source/source_library_builder.dart
+++ b/pkg/front_end/lib/src/fasta/source/source_library_builder.dart
@@@ -3422,7 -3573,11 +3422,15 @@@ class SourceLibraryBuilder extends Libr
        DartType superBoundedAttempt,
        DartType superBoundedAttemptInverted}) {
      List<LocatedMessage> context;
++<<<<<<< HEAD
 +    if (typeParameter != null && typeParameter.fileOffset != -1) {
++=======
+     // Skip reporting location for function-type type parameters as it's a
+     // limitation of Kernel.
+     if (typeParameter != null &&
+         typeParameter.fileOffset != -1 &&
+         typeParameter.location != null) {
++>>>>>>> 4c1a153c090 ([cfe] Don't show context on type argument issues if location is null)
        // It looks like when parameters come from patch files, they don't
        // have a reportable location.
        (context ??= <LocatedMessage>[]).add(
```

@johnniwinther @mraleph Can you provide merge instructions?

**Comment by mraleph** [comment]
@athomas simply take the lower part (between `======` and `>>>>>>>`)

**Comment by athomas** [comment]
Merged to stable in 065872e802209ffa97aa3fad204758830b6100a0 (2.13.4).
